test/extensions/filters/http/ext_authz/ext_authz_grpc_fuzz_test.cc

```diff
@@ -0,0 +1,128 @@
+#include "envoy/extensions/filters/http/ext_authz/v3/ext_authz.pb.validate.h"
+
+#include "source/common/http/context_impl.h"
+#include "source/common/network/address_impl.h"
+#include "source/extensions/filters/http/ext_authz/ext_authz.h"
+
+#include "test/extensions/filters/common/ext_authz/mocks.h"
+#include "test/extensions/filters/common/ext_authz/test_common.h"
+#include "test/extensions/filters/http/common/fuzz/http_filter_fuzzer.h"
+#include "test/extensions/filters/http/ext_authz/ext_authz_fuzz.pb.validate.h"
+#include "test/extensions/filters/http/ext_authz/ext_authz_fuzz_lib.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/grpc/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/server/server_factory_context.h"
+#include "test/mocks/upstream/cluster_manager.h"
+
+#include "gmock/gmock.h"
+
+using Envoy::Extensions::Filters::Common::ExtAuthz::TestCommon;
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExtAuthz {
+namespace {
+
+std::unique_ptr<envoy::service::auth::v3::CheckResponse>
+makeGrpcCheckResponse(const Grpc::Status::WellKnownGrpcStatus status) {
+  auto response = std::make_unique<envoy::service::auth::v3::CheckResponse>();
+  response->mutable_status()->set_code(status);
+  // TODO: We only add the response status.
+  // Add fuzzed inputs for headers_to_(set/append/add), body, status_code to the Response.
+  return response;
+}
+
+Grpc::Status::WellKnownGrpcStatus resultCaseToGrpcStatus(
+    const envoy::extensions::filters::http::ext_authz::ExtAuthzTestCase::AuthResult result) {
+  Grpc::Status::WellKnownGrpcStatus check_status;
+  switch (result) {
+  case envoy::extensions::filters::http::ext_authz::ExtAuthzTestCase::OK: {
+    check_status = Grpc::Status::WellKnownGrpcStatus::Ok;
+    break;
+  }
+  case envoy::extensions::filters::http::ext_authz::ExtAuthzTestCase::ERROR: {
+    check_status = Grpc::Status::WellKnownGrpcStatus::Internal;
+    break;
+  }
+  case envoy::extensions::filters::http::ext_authz::ExtAuthzTestCase::DENIED: {
+    check_status = Grpc::Status::WellKnownGrpcStatus::PermissionDenied;
+    break;
+  }
+  default: {
```

Comment:
Thanks for working on this!
Just a quick drive-by comment. If you do PANIC here, you probably want to remove default case, and use exhaustive switch case.  PANIC_ON_PROTO_ENUM_SENTINEL_VALUES/PANIC_DUE_TO_CORRUPT_ENUM

Reply:
TIL! Done.